### ISSUE TITLE
[BUG] postgres_extension.present fails to detect extention in test mode

### ISSUE THREAD

**Issue by 4s1** [saltstack/salt#38966]
### Description of Issue
Running state.apply in testmode detects a given postgres-extension as missing even if it is installed. Actually running state.apply on the other hand, won't change anything, as it detects the postgres-extension correctly as installed.
 

### Setup

postgres.sls
````
gitlab-database:
[...]
  postgres_extension.present:
    - name: pg_trgm
    - maintenance_db: gitlab
    - require:
        - postgres_database: gitlab
[...]
````

### Steps to Reproduce Issue
1. configure a minion to host a database (gitlab) and install an (pg_trgm) extension into this database (either manual or by tusing the given state artifact)
2. query this minion, if the extension (pg_trgm) is installed in the database: `salt 'minion' postgres.is_installed_extension name=pg_trgm maintenance_db=database`, the returned result should be _True_
3. run `salt 'minion' state.apply` which should report no changes
3. now run the same command again in test mode: `salt 'minion' state.apply test=True` and see, it reports the postgres extension as missing

### Versions Report
Salt Version:
           Salt: 2016.11.1
 
Dependency Versions:
           cffi: 1.9.1
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: 0.24.3
        libnacl: Not Installed
       M2Crypto: 0.24.0
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
         pygit2: 0.24.2
         Python: 2.7.13 (default, Dec 21 2016, 07:16:46)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.2.0
 
System Versions:
           dist:   
        machine: x86_64
        release: 4.7.4-1-ARCH
         system: Linux
        version: Not Installed

**Comment by gtmanfred** [comment]
Is the gitlab plugin available publically somewhere so I can test it?  I tried with the citusdb plugin and it is working in 2016.11.1

```
[root@salt ~]# salt-call --local state.apply test test=True
local:
----------
          ID: citus
    Function: postgres_extension.present
      Result: None
     Comment: Extension citus is set to be installed
     Started: 15:52:24.804679
    Duration: 58.354 ms
     Changes:

Summary for local
------------
Succeeded: 1 (unchanged=1)
Failed:    0
------------
Total states run:     1
Total run time:  58.354 ms
[root@salt ~]# salt-call --local state.apply test
local:
----------
          ID: citus
    Function: postgres_extension.present
      Result: True
     Comment: The extension citus has been installed
     Started: 15:52:29.286227
    Duration: 370.425 ms
     Changes:
              ----------
              citus:
                  Installed

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time: 370.425 ms
[root@salt ~]# salt-call --local state.apply test
local:
----------
          ID: citus
    Function: postgres_extension.present
      Result: True
     Comment: Extension citus is already present
     Started: 15:52:33.168921
    Duration: 56.718 ms
     Changes:

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:  56.718 ms
```

```
[root@salt ~]# cat /srv/salt/test.sls
citus:
  postgres_extension.present
```

Thanks,
Daniel

**Comment by 4s1** [comment]
I think pg_trgm is bundled with postgres by default.

Here's some additional info from running `salt-minion -l debug` on database server.

/srv/salt/dev/states/posgtres/pdb01.sls
````
pg_trgm:
  postgres_extension.present
````

`salt 'pdb01*' state.apply --state-output=changes --state-verbose=False saltenv=dev pillarenv=dev test=True postgres.pdb01`
````
[INFO    ] Running state [pg_trgm] at time 20:49:52.826529
[INFO    ] Executing state postgres_extension.present for pg_trgm
[INFO    ] Executing command ['/usr/bin/psql', '--no-align', '--no-readline', '--no-psqlrc', '--no-password', '--dbname', 'postgres', '-v', 'datestyle=ISO,MDY', '-c', 'COPY (select a.*, b.nspname as schema_name from pg_extension a,  pg_namespace b where a.extnamespace = b.oid) TO STDOUT WITH CSV HEADER'] as user 'postgres' in directory '/var/lib/postgres'
[DEBUG   ] stdout: extname,extowner,extnamespace,extrelocatable,extversion,extconfig,extcondition,schema_name
plpgsql,10,11,f,1.0,,,pg_catalog
pg_trgm,10,2200,t,1.3,,,public
[INFO    ] Extension pg_trgm is set to be created
````

`salt 'pdb01*' state.apply --state-output=changes --state-verbose=False saltenv=dev pillarenv=dev postgres.pdb01`
````
[INFO    ] Running state [pg_trgm] at time 20:50:12.672342
[INFO    ] Executing state postgres_extension.present for pg_trgm
[INFO    ] Executing command ['/usr/bin/psql', '--no-align', '--no-readline', '--no-psqlrc', '--no-password', '--dbname', 'postgres', '-v', 'datestyle=ISO,MDY', '-c', 'COPY (select a.*, b.nspname as schema_name from pg_extension a,  pg_namespace b where a.extnamespace = b.oid) TO STDOUT WITH CSV HEADER'] as user 'postgres' in directory '/var/lib/postgres'
[DEBUG   ] stdout: extname,extowner,extnamespace,extrelocatable,extversion,extconfig,extcondition,schema_name
plpgsql,10,11,f,1.0,,,pg_catalog
pg_trgm,10,2200,t,1.3,,,public
[INFO    ] Extension pg_trgm is already present
````

**Comment by blaketmiller** [comment]
I've got this issue as well. Including my relevant debug code:

```
[DEBUG   ] LazyLoaded postgres_extension.present
[INFO    ] Running state [pldbgapi] at time 16:02:37.232678
[INFO    ] Executing state postgres_extension.present for pldbgapi
[INFO    ] Executing command ['/usr/pgsql-9.6/bin/psql', '--no-align', '--no-readline', '--no-psqlrc', '--no-password', '--dbname', 'postgres', '-v', 'datestyle=ISO,MDY', '-c', 'COPY (select a.*, b.nspname as schema_name from pg_extension a,  pg_namespace b where a.extnamespace = b.oid) TO STDOUT WITH CSV HEADER'] as user 'postgres' in directory '/var/lib/pgsql'
[DEBUG   ] stdout: extname,extowner,extnamespace,extrelocatable,extversion,extconfig,extcondition,schema_name
plpgsql,10,11,f,1.0,,,pg_catalog
pldbgapi,10,2200,t,1.0,,,public
[INFO    ] Extension pldbgapi is set to be created
```
Extracting out the command and running it manually yields the same results:
```
$ psql --no-align --no-readline --no-psqlrc --no-password --dbname postgres -v datestyle=ISO,MDY -c 'COPY (select a.*, b.nspname as schema_name from pg_extension a,  pg_namespace b where a.extnamespace = b.oid) TO STDOUT WITH CSV HEADER'
extname,extowner,extnamespace,extrelocatable,extversion,extconfig,extcondition,schema_name
plpgsql,10,11,f,1.0,,,pg_catalog
pldbgapi,10,2200,t,1.0,,,public
```

**Comment by gtmanfred** [comment]
Sorry, i completely forgot about this issue.

I will take a look tomorrow.

Thanks,
Daniel

**Comment by gtmanfred** [comment]
Actually, after looking closer at this issue, i miss understood what was going on.

Looking at the code, it does look like we just need a return at the top of the function saying that the extension is already installed.

I have tagged this as a medium severity issue because it is mostly cosmetic.  If anyone wanted to submit a PR to fix this, we would greatly appreciate it, otherwise I have tagged it.

Thanks,
Daniel
